lib/mshoplib/setup/TablesUpdateCharsetCollation.php

```diff
@@ -171,6 +171,40 @@ protected function checkColumns( \Aimeos\MW\Setup\DBSchema\Iface $schema, $table
 			return true;
 		}
 
+		return false;
+	}
+
+    /**
+	 * Checking the MySql compatibility.
+	 *
+	 * @param \Aimeos\MW\Setup\DBSchema\Iface $schema
+	 * @return bool
+	 */
+	protected function checkMySqlCompatibility( \Aimeos\MW\Setup\DBSchema\Iface $schema )
+	{
+		if ( !$schema instanceof \Aimeos\MW\Setup\DBSchema\Mysql )
+		{
+			return true;
+		}
+
+		// Ask the database for it's type and version.
+		// A MariaDB gets identified as a MySql 5.5.5 by doctrine.
+		// We can not rely on it to tell us the actual version and ask
+		// the server directly.
+		$conn = $this->acquire();
+		$result = $conn->create( 'SELECT version()' )->execute();
+		// Something like '10.1.29-MariaDB' or '5.6.33-0ubuntu0...'
+		$version = $result->fetch();
+		$version = $version['version()'];
+		$result->finish();
+		$this->release($conn);
+
+		if( ( strpos( $version, 'MariaDB' ) !== false && version_compare( $version, '10.2', '>=' ) ) ||
+			version_compare( $version, '5.7', '>=' )
```

Comment:
Do you need to check for `strpos( $version, 'MySQL' ) !== false` too?

Reply:
'!== false' means it is somewhere in there.

The string may look like '10.1.29-MariaDB'. The distro might also add stuff at the end.

Reply:
The MySql does not claim to be a MySql in its version number.
The dev system I'm currently working with replies with '5.6.33-0ubuntu0.14.04.1'

Reply:
Then your check is OK :-)